cdap-explore/src/main/java/co/cask/cdap/explore/executor/ExploreExecutorHttpHandler.java

```diff
@@ -180,18 +185,60 @@ public void enableDataset(HttpRequest request, HttpResponder responder,
     try {
       // Note: Namespacing will come later.
       Id.DatasetInstance datasetInstanceId = Id.DatasetInstance.from(Constants.DEFAULT_NAMESPACE, datasetName);
+      DatasetSpecification datasetSpec = datasetFramework.getDatasetSpec(datasetInstanceId);
+      if (datasetSpec == null) {
+        responder.sendString(HttpResponseStatus.NOT_FOUND, "Cannot load dataset " + datasetInstanceId);
+        return;
+      }
+
+      String createStatement = null;
+      // some datasets cannot be instantiated here. For example, ObjectMappedTable is often parameterized with a type
+      // that is only available in a program context and not available here in the system context.
+      // explore should only have logic related to exploration and not dataset logic.
+      // TODO: refactor exploration (CDAP-1573)
+      String datasetType = datasetSpec.getType();
+      // special casing here... but we really should clean this up
+      // there are two ways to refer to each dataset type...
+      if (ObjectMappedTableModule.FULL_NAME.equals(datasetType) ||
+        ObjectMappedTableModule.SHORT_NAME.equals(datasetType)) {
+        // ObjectMappedTable must contain a schema in its properties
+        String schemaStr = datasetSpec.getProperty(DatasetProperties.SCHEMA);
+        if (schemaStr == null) {
+          responder.sendString(HttpResponseStatus.NOT_FOUND, "Schema not found for dataset " + datasetInstanceId);
```

Comment:
Seems that this is more of a bad request than a not found. Because the dataset itself exists, but its properties are incomplete